pkg/chunk/chunk_store.go

```diff
@@ -205,15 +205,20 @@ func (c *store) LabelValuesForMetricName(ctx context.Context, userID string, fro
 	}
 
 	var result []string
+	values := map[string]struct{}{}
 	for _, entry := range entries {
 		_, labelValue, _, _, err := parseChunkTimeRangeValue(entry.RangeValue, entry.Value)
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, string(labelValue))
+		v := string(labelValue)
+		if _, ok := values[v]; !ok {
```

Comment:
I guess you can use `result = appendUniqueStrings(result, values, []string{v})` or you see it inefficient?

Reply:
I was actually trying to find a way to use it here, there's another place also and no I think it is a good idea.